### PR TITLE
camelCase -> snake_case

### PR DESCRIPTION
**Pull request by pr0n00gler**

### CODE REVIEW

**Review by swelf19 — COMMENTED** [review]
pls fix the json tag

File: proto/neutron/dex/deposit_record.proto [hunk 1]
```diff
@@ -6,15 +6,15 @@ import "gogoproto/gogo.proto";
...
-  PairID pairID = 1;
-  string sharesOwned = 2 [
-                          (gogoproto.moretags)   = "yaml:\"totalShares\"",
+  PairID pair_id = 1;
+  string shares_owned = 2 [
+                          (gogoproto.moretags)   = "yaml:\"total_shares\"",
                           (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
                           (gogoproto.nullable)   = false,
-                          (gogoproto.jsontag) = "totalShares"
+                          (gogoproto.jsontag) = "total_shares"
```
Comment (swelf19) [inline proto/neutron/dex/deposit_record.proto]: json tag differs from a field name

Comment (swelf19) [inline proto/neutron/dex/deposit_record.proto]:
```suggestion
                          (gogoproto.jsontag) = "shares_owned"
```

File: proto/neutron/dex/deposit_record.proto [hunk 2]
```diff
@@ -6,15 +6,15 @@ import "gogoproto/gogo.proto";
 import "neutron/dex/pair_id.proto";
 
 message DepositRecord {
-  PairID pairID = 1;
-  string sharesOwned = 2 [
-                          (gogoproto.moretags)   = "yaml:\"totalShares\"",
+  PairID pair_id = 1;
+  string shares_owned = 2 [
+                          (gogoproto.moretags)   = "yaml:\"total_shares\"",
```
Comment (swelf19) [inline proto/neutron/dex/deposit_record.proto]:
```suggestion
                          (gogoproto.moretags)   = "yaml:\"shares_owned\"",
```

File: proto/neutron/dex/limit_order_tranche_user.proto [hunk 3]
```diff
@@ -7,28 +7,28 @@ import "neutron/dex/trade_pair_id.proto";
...
+  int64 tick_index_taker_to_maker = 2;
+  string tranche_key = 3;
   string address = 4; 
-  string sharesOwned = 5  [
-      (gogoproto.moretags)   = "yaml:\"sharesOwned\"",
+  string shares_owned = 5  [
+      (gogoproto.moretags)   = "yaml:\"shares_owned\"",
       (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
       (gogoproto.nullable)   = false,
       (gogoproto.jsontag) = "sharesOwned"
```
Comment (swelf19) [inline proto/neutron/dex/limit_order_tranche_user.proto]:
```suggestion
      (gogoproto.jsontag) = "shares_owned"
```

**Review by swelf19 — APPROVED** [review]

**Review by jcompagni10 — APPROVED** [review]

**Pull request merged**